### PR TITLE
Use Azure to build the MSI installer

### PR DESCRIPTION
**Pull request by rcurtin**
For #2458, it seems like we'll have to give up on using AppVeyor, and in order to do that, we need to transfer the last thing that we're using AppVeyor for---the MSI installer.  So, this PR will probably make many attempts to get that working... :)

### CODE REVIEW

**Comment by zoq** [comment]
I did some work on the MSI installer some time ago, I can see if I can find it.

**Comment by rcurtin** [comment]
Oh, awesome, maybe you have gotten farther than I did. :smile:  I can take the work from wherever you had it and keep fighting. :)

**Comment by zoq** [comment]
Maybe, here is the last version: https://github.com/zoq/mlpack/blob/ea26c88f0fce8fb6c2d77eee395b5cffbd7a15aa/.ci.yaml

https://github.com/zoq/mlpack/commits/azure-pl here is the branch - at some point I thought well let's use the AppVeyor job to get the MSI package.

**Comment by zoq** [comment]
The commit messages are the best :)

**Comment by rcurtin** [comment]
Ok, I can't believe it, but I actually got everything working here so it's ready for review!  After this is merged, we will be able to remove our AppVeyor job entirely.  In order to make all of it work, I followed this repository: https://github.com/DavidEGrayson/wix-example-harvest-directory/blob/master/foo.wixproj

I'm not sure why the existing configuration didn't port over nicely, but after having put two straight months into this PR, I'm not keen on going back and figuring out why. :smile:

File: .ci/windows-steps.yaml [hunk 1]
```diff
@@ -118,6 +118,61 @@ steps:
...
+    }
+    catch{Write-Output "Unable to add doc to installer, skipping!"}
+    # Preparing installer staging.
+    mkdir dist\win-installer\mlpack-win-installer\Sources\lib
+    cp build\Release\*.lib dist\win-installer\mlpack-win-installer\Sources\lib\
+    cp build\Release\*.exp dist\win-installer\mlpack-win-installer\Sources\lib\
+    cp build\Release\*.dll dist\win-installer\mlpack-win-installer\Sources\
+    cp build\Release\*.exe dist\win-installer\mlpack-win-installer\Sources\
+    cp $(Agent.ToolsDirectory)\OpenBLAS.0.2.14.1\lib\native\bin\x64\*.dll dist\win-installer\mlpack-win-installer\Sources\
+    cp $(Agent.ToolsDirectory)\boost_libs\boost_unit_test_framework-vc*.dll dist\win-installer\mlpack-win-installer\Sources\
```
Comment (shrit) [inline .ci/windows-steps.yaml:143]:
```suggestion
    cp $(Agent.ToolsDirectory)\boost_libs\boost_unit_test_framework-vc*.dll dist\win-installer\mlpack-win-installer\Sources\
```
I am not sure we still need this line

**Review by shrit — COMMENTED** [review]
Everything looks fine, just a minor comment, also I am not an expert in Azure, and I have very limited knowledge in Windows.

File: .ci/windows-steps.yaml (hunk 1 above)
Comment (rcurtin) [reply .ci/windows-steps.yaml]: I think you're right---thanks for pointing that out!  I removed it in 1ae1bae7a.  Let's see if everything still works. :)

File: .ci/windows-steps.yaml [hunk 2]
```diff
@@ -118,6 +116,60 @@ steps:
...
+    else
+    {
+      $env:INSTALL_VERSION = $env:MLPACK_VERSION;
+    }
+
+    # Build the MSI installer.
+    cd dist\win-installer\mlpack-win-installer
+    dir
+    cd Sources
+    dir
```
Comment (zoq) [inline .ci/windows-steps.yaml:164]: Do we like to keep the `dir` here for future debug sessions, or can we remove them?

**Review by zoq — COMMENTED** [review]
Wow, this is awesome 👍

**Review by shrit — APPROVED** [review]

File: .ci/windows-steps.yaml (hunk 2 above)
Comment (rcurtin) [reply .ci/windows-steps.yaml]: Good point, thanks!  Removed in 36abf9577.

**Review by zoq — APPROVED** [review]
This is awesome!

**Comment by rcurtin** [comment]
Great, now I will remove the AppVeyor integration since we no longer need it. :+1:

**Pull request merged**